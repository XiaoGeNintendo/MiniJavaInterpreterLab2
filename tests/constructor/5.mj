class Fa{
    int fa=-2;
}
class X extends Fa{
    int d=this.fa+3;
    int a=this.x+3;
    int z=this.x+3;
    int x=3;
    int y=this.x+3;
    int c=this.b+=3;
    int b=0;
    int w=this.k+=3;
    int k;
    X(){
        this(1);
        println(d+" "+x+" "+y+" "+z+" "+a+" "+c+" "+b+" "+w+" "+k);
    }

    X(int _){
        this(1,2);
        println(d+" "+x+" "+y+" "+z+" "+a+" "+c+" "+b+" "+w+" "+k);
    }

    X(int _,int __){
        println(d+" "+x+" "+y+" "+z+" "+a+" "+c+" "+b+" "+w+" "+k);
    }
}

int main(){
    new X();
    return 0;
}