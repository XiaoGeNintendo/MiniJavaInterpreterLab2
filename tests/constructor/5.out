1 3 6 3 3 3 0 3 3
1 3 6 3 3 3 0 3 3
1 3 6 3 3 3 0 3 3
Process exits with the code 0.